### DISCUSSION TITLE
[bug] <a>+<img> with .gif images are broken

### DISCUSSION THREAD

**Discussion by ZacharyPatten** [community/community#17684] [Other Feature Feedback, Questions, & Ideas]
> I submitted a ticket <sub>#1632354</sub> to GitHub's support for this topic, but my ticket was closed and I was encouraged to post the bug here instead.

## Bug Description

If you use a `.gif` image in an `<img>` tag in markdown files wrapped in a `<a>` tag the link provided by the `<a>` is lost, so when you click on the rendered image, it will not take you to the link.

## Example

Markdown:
```md
<p align="center">
	<a href="https://github.com/ZacharyPatten/dotnet-console-games/blob/main/.github/resources/animated-banner-credit.md"><img width="600" src="https://github.com/ZacharyPatten/dotnet-console-games/blob/main/.github/resources/animated-banner.gif?raw=true" ></a>
</p>
```
<sup>source of this example is from the root `README.md` file in this repo: https://github.com/ZacharyPatten/dotnet-console-games</sup>

HTML Rendered by GitHub for above Markdown:
```html
<p align="center" dir="auto">
	<animated-image data-catalyst=""><a href="https://github.com/ZacharyPatten/dotnet-console-games/blob/main/.github/resources/animated-banner-credit.md" data-target="animated-image.originalLink" hidden=""><img width="600" src="https://github.com/ZacharyPatten/dotnet-console-games/raw/main/.github/resources/animated-banner.gif?raw=true" style="max-width: 100%;" data-target="animated-image.originalImage" hidden=""></a>
      <span class="AnimatedImagePlayer" data-target="animated-image.player">
        <a data-target="animated-image.replacedLink" class="AnimatedImagePlayer-images" href="https://github.com/ZacharyPatten/dotnet-console-games/blob/main/.github/resources/animated-banner-credit.md" target="_blank" hidden="">
          
        </a>
        <button data-target="animated-image.imageButton" class="AnimatedImagePlayer-images" tabindex="-1" aria-label="Play animated-banner.gif?raw=true"><span data-target="animated-image.imageContainer">
            <img data-target="animated-image.replacedImage" alt="animated-banner.gif?raw=true" class="AnimatedImagePlayer-animatedImage" src="https://github.com/ZacharyPatten/dotnet-console-games/raw/main/.github/resources/animated-banner.gif?raw=true" width="600" style="display: block; opacity: 1;">
          <canvas class="AnimatedImagePlayer-stillImage" aria-hidden="true" width="600" height="338"></canvas></span></button>
        <span class="AnimatedImagePlayer-controls" data-target="animated-image.controls">
          <button data-target="animated-image.playButton" class="AnimatedImagePlayer-button" aria-label="Play animated-banner.gif?raw=true">
            <svg aria-hidden="true" focusable="false" class="octicon icon-play" width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
              <path d="M4 13.5427V2.45734C4 1.82607 4.69692 1.4435 5.2295 1.78241L13.9394 7.32507C14.4334 7.63943 14.4334 8.36057 13.9394 8.67493L5.2295 14.2176C4.69692 14.5565 4 14.1739 4 13.5427Z">
            </path></svg>
            <svg aria-hidden="true" focusable="false" class="octicon icon-pause" width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
              <rect x="4" y="2" width="3" height="12" rx="1"></rect>
              <rect x="9" y="2" width="3" height="12" rx="1"></rect>
            </svg>
          </button>
          <a data-target="animated-image.openButton" aria-label="Open animated-banner.gif?raw=true in new window" class="AnimatedImagePlayer-button" href="https://github.com/ZacharyPatten/dotnet-console-games/blob/main/.github/resources/animated-banner-credit.md" target="_blank">
            <svg aria-hidden="true" class="octicon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16">
              <path fill-rule="evenodd" d="M10.604 1h4.146a.25.25 0 01.25.25v4.146a.25.25 0 01-.427.177L13.03 4.03 9.28 7.78a.75.75 0 01-1.06-1.06l3.75-3.75-1.543-1.543A.25.25 0 0110.604 1zM3.75 2A1.75 1.75 0 002 3.75v8.5c0 .966.784 1.75 1.75 1.75h8.5A1.75 1.75 0 0014 12.25v-3.5a.75.75 0 00-1.5 0v3.5a.25.25 0 01-.25.25h-8.5a.25.25 0 01-.25-.25v-8.5a.25.25 0 01.25-.25h3.5a.75.75 0 000-1.5h-3.5z"></path>
            </svg>
          </a>
        </span>
      </span></animated-image>
</p>
```

As you can see, the `<span class="AnimatedImagePlayer" data-target="animated-image.player">` that is generated is not inside the `<a>` tag so it no longer acts as a link. This is a bug.

## Timing

This issue was not occurring back in March 2022. So it is likely that a code change since March 2022 has introduced this bug.

**Comment by ZacharyPatten** [comment]
Update: I just checked this morning and this bug appears to have been fixed, so it looks like another code change must have been pushed to fix the issue. Hopefully some unit testing was added to ensure it will stay fixed. :D

**Comment by TaylanTatli** [comment]
I have this problem right now with my profile readme.

**Comment by ZacharyPatten** [comment]
Now `.gif` images are causing pages to shrink on mobile...

<img src="https://user-images.githubusercontent.com/3385986/180123835-515f95d5-b073-408e-b654-80c9d31e67e1.png" height="300" />

They keep introducing bugs with `.gif` images... 🤦‍♂️ stahp

**Reply by gerrywastaken** [reply]
.gifs in READMEs are the bug

**Comment by Anoerak** [comment]
This issue seem originate from the file `syntax-theme-codemirror.scss`and the rule `[data-target="animated-image.originalImage"], [data-target="animated-image.replacedImage"], [data-a11y-animated-images=system] [data-animated-image], [data-a11y-animated-images=disabled] [data-animated-image] `which commands a `display: none`.
Don't know how to fix it though..

**Comment by Anoerak** [comment]
Ok, got it.
You need to go  to [Settings/Accessibility](https://github.com/settings/accessibility), then scroll down to `Motion` and click `enabled` to force the Gif to animated all the time.

**Comment by Mustafa-Hassan2001** [comment]
The issue described occurs because Markdown does not support clickable images with links when using both the <a> and <img> tags.

**Comment by laggykiller** [comment]
Similar bug occurs when gif image in table of markdown file. the gif clips through the top bar.

As suggested above, if 'Autoplay animated images' is enabled then the bug does not occurs, if disabled then the bug occurs.

See https://github.com/orgs/community/discussions/83639
